runtimes/common/src/constants.rs

```diff
@@ -324,7 +324,18 @@ pub mod did {
 	/// The size is checked in the runtime by a test.
 	pub const MAX_DID_BYTE_LENGTH: u32 = 9918;
 
-	pub const DID_DEPOSIT: Balance = deposit(2 + MAX_NUMBER_OF_SERVICES_PER_DID, MAX_DID_BYTE_LENGTH);
+	/// The length of all used Keys.
+	pub const KEY_DID_BYTE_LENGTH: u32 = 32;
+
+	pub const MAX_SERVICE_ENDPOINT_BYTE_LENGTH: u32 =
+		MAX_SERVICE_URL_LENGTH + MAX_SERVICE_ID_LENGTH + MAX_SERVICE_TYPE_LENGTH;
+
+	pub const DID_BASE_DEPOSIT: Balance = 2 * KILT;
+	pub const KEY_DEPOSIT: Balance = deposit(0, KEY_DID_BYTE_LENGTH);
+	pub const SERVICE_ENDPOINT_DEPOSIT: Balance = deposit(1, MAX_SERVICE_ENDPOINT_BYTE_LENGTH);
+	pub const MAX_DEPOSIT_DID: Balance =
```

Comment:
Is this accurate? Do we actually have 2 + number of services elements, each of size `MAX_DID_BYTE_LENGTH`?

Reply:
The value MAX_DEPOSIT_DID consists of the old did deposit, which was the worst case, plus the new base deposit. 

The max byte length and the amount of storage entries is still the same.

Reply:
It should be accurate, since it is the old deposit + the new base deposit. MAX_DID_BYTE_LENGTH is the worst case of a single DID. We have 27 different storage entries. 1 for the did_details and 1 for the service endpoint count, if I am not wrong 😃

Reply:
Yes, I see that now only the base deposit is added, which looks correct. I am just wondering about the nature of the `deposit` function:

```
/// Calculate the storage deposit based on the number of storage items and the
/// combined byte size of those items.
```

I understood it that for each item type, we would count how much space they take and how many entries we write in storage. But reading it again now, it says **combined**, so I guess items are to be counted separately, but the grand total is the total size. In that case, this is correct ☺️